crowd_sim/envs/crowd_sim.py

```diff
@@ -77,10 +87,39 @@ def configure(self, config):
             logging.info("Not randomize human's radius and preferred speed")
         logging.info('Training simulation: {}, test simulation: {}'.format(self.train_val_sim, self.test_sim))
         logging.info('Square width: {}, circle width: {}'.format(self.square_width, self.circle_radius))
+        logging.info("Number of static obstacles: {}".format(self.static_obstacle_num))
+        logging.info("Static obstacles diameter range: {} - {}".format(2 * self.obstacle_min_radius, 2 * self.obstacle_max_radius))
 
     def set_robot(self, robot):
         self.robot = robot
 
+    def generate_random_obstacles(self, obs_num):
+        width = self.square_width
+        height = self.square_width
+        max_radius = self.obstacle_max_radius - self.obstacle_min_radius
+        min_radius = self.obstacle_min_radius
+        self.obs = []
+        # if obs_num == 0:
+        #     human = Human(self.config, 'humans')
+        #     human.set(0, -10, 0, -10, 0, 0, 0)
+        #     self.obs.append(human)
+        for i in range(obs_num):
+            human = Human(self.config, 'humans') ## we model the static obstacles as static humans
+            while True:
+                px = (np.random.random() - 0.5) * width
+                py = (np.random.random() - 0.5) * height
+                r = (np.random.random()) * max_radius + min_radius
+                collide = False
+                for agent in [self.robot] + self.obs:
+                    if norm((px - agent.px, py - agent.py)) < r + agent.radius + self.discomfort_dist + self.obstacle_width or norm((px - self.robot_gx, py - self.robot_gy)) < r + self.discomfort_dist:
+                        collide = True
+                        break
+                if not collide:
+                    break
+            human.set(px, py, px, py, 0, 0, 0, radius=r)
+            # print("Generate obstacle!")
+            self.obs.append(human)
```

Comment:
I totally get that! I actually think about it earlier but by then i just want to see the result as soon as possible lol. I think this would be a great suggestion. Thanks! Will work on it later